src/server/frontend_wayland/wl_pointer.cpp

```diff
@@ -232,9 +232,22 @@ void mf::WlPointer::axis(MirPointerEvent const* event)
         break;
 
     case mir_pointer_axis_source_wheel:
+    {
         send_axis_source_event(AxisSource::wheel);
+
+        if (auto const h_scroll_discrete = mir_pointer_event_axis_value(event, mir_pointer_axis_hscroll_discrete))
+        {
+            send_axis_discrete_event(Axis::horizontal_scroll, h_scroll_discrete);
+        }
```

Comment:
1. Is it wise to only send discrete events for `axis_source_wheel`? I believe our platform will only add discrete scroll to that source, but the protocol doesn't forbid it for other sources. I see no reason to prevent other sources from sending discrete events here.
2. The protocol is very specific that:
    > This event does not occur on its own, it is coupled with wl_pointer.axis event
  
    in general we will send both, but if for some reason the platform gives us discrete scroll with 0 normal scroll, we will not and thus make a protocol violation.